### PR TITLE
Convert to @types

### PR DESCRIPTION
**Pull request by kitsonk**
**Type:** feature

**Related Issue:** dojo/meta#46

Please review this checklist before submitting your PR:
- [X] There is a related issue
- [X] All contributors have signed a CLA
- [X] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
- [X] The code passes the CI tests
- [X] Unit or Functional tests are included in the PR
- [X] The PR increases or maintains the overall unit test coverage percentage
- [X] The code is ready to be merged

Migrates to using `@types` where possible, changes to `moduleResolution=node`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,24 +24,15 @@
...
-    "grunt-ts": "^5.5.1",
-    "grunt-tslint": "^3.0.3",
-    "grunt-typings": ">=0.1.5",
     "intern": "~3.2.3",
-    "istanbul": "0.4.3",
-    "remap-istanbul": ">=0.6.4",
-    "tslint": "next",
-    "typescript": "2.0.2"
+    "tslint": "^3.15.1",
+    "typescript": "~2.0.3"
```
Comment (agubler) [inline package.json:36]: tilde not caret?

Comment (kitsonk) [reply package.json:36]: I suspect 2.1 will introduce breaking changes, contrary to semver... I realised that we should do this with the others too, don't want this accidentally flipping over to 2.1.

**Comment by kitsonk** [comment]
I also converted this to TS2.0 Latest realised that a lot of the `devDependencies` are brought in (and therefore should be version managed) by `grunt-dojo2`.  I will reflect these in `dojo2-package-template` as well.

File: package.json (hunk 1 above)
Comment (agubler) [reply package.json:36]: okay, I guess that's probably a fair assumption (although it really shouldn't, oh well)

**Review by agubler — APPROVED** [review]

**Pull request merged**